tests/playwright/Report.md

```diff
@@ -0,0 +1,295 @@
+- [Puppeteer Evaluation](https://www.sohamp.dev/blog/2020-08-24-wmf-gsoc2020/) by Soham Parekh
+
+So, this brings us to the current framework being evaluated against WebdriverIO, which is Playwright. The evaluations progress involved the following steps;
+
+- Configuring and Setting up of Wikimedia Core to run on CI, in this case on Github actions.
+- Implementation of all MediaWiki Core tests in Playwright.
+- Running MediaWiki Core tests in WebdriverIO.
+- Analysis of both WebdriverIO and Playwright tests in terms of speed and stability.
+
+## WebdriverIO
+
+WebdriverIO is a test automation framework that allows you to run tests based on the Webdriver protocol and Appium automation technology. WebdriverIO is known to be;
+
+- Extendable: Adding helper functions, or more complicated sets and combinations of existing commands is simple and really useful
+- Compatible: WebdriverIO can be run on the WebDriver Protocol for true cross-browser testing as well as Chrome DevTools Protocol for Chromium-based automation using Puppeteer.
+- Feature Rich: The huge variety of community plugins allows you to easily integrate and extend your setup to fulfill your requirements.
+
+The above points are just a few of WebdriverIO capability. To explore more WebdriverIO capabilities, you can check out the WebdriverIO and [MediaWiki documentation](https://www.mediawiki.org/wiki/Selenium).
+
+WebdriverIO provides the ability and options to run commands in both asynchronous and synchronous operations. For asynchronous operations, you can use the JavaScript usual async/await, however for synchronous, it can be done through [node-fibers](https://www.npmjs.com/package/fibers). Both of these options have their own benefits and issues which can be found on the [WebdriverIO](https://webdriver.io/docs/sync-vs-async) website.
+
+### Sample WebdriverIO Code
+
+This code snippet navigates to wikipedia.org in Chrome and saves the screenshot.
+
+```js
+const { remote } = require('webdriverio');
+
+(async () => {
+ const browser = await remote({
+        capabilities: {
+            browserName: 'chrome'
+        }
+  })
+  await browser.url('https://www.wikipedia.org/')
+
+  await browser.getTitle()
+
+  await browser.saveScreenshot('./screenshot.png')
+  await browser.deleteSession();
+})();
+```
+
+### Mocha
+
+The WebdriverIO WDIO runner currently supports Mocha, Jasmine, and Cucumber frameworks and WikiMedia Core test is currently using Mocha framework. [Mocha](https://mochajs.org/) is a feature-rich JavaScript test framework running on Node.js and in the browser, making asynchronous testing simple and fun.
+
+WebdriverIO provides an adapter for Mocha which can easily be added as `@wdio/mocha-framework`.
+
+#### Sample Mocha Code
+
+The code snippet below navigates to wikipedia.org in Chrome and checks if the correct title is being displayed
```

Comment:
To be a useful comparison, wdio and mocha code samples should do the same thing, but they are not. wdio code is opening a page, getting it's title (and then not doing anything with it) and taking a screenshot. mocha example is opening a page and asserting that the title is correct, but not taking a screenshot.

I suggest to simplify and make both code samples open a page and check if the title is correct (not taking the screenshot).